app/controllers/invites_controller.rb

```diff
@@ -251,11 +251,19 @@ def perform_accept_invitation
       topic = invite.topics.first
       response = {}
 
-      if user.present? && user.active?
-        response[:redirect_to] = topic.present? ? path(topic.relative_url) : path("/")
-      elsif user.present?
-        response[:message] = I18n.t('invite.confirm_email')
-        cookies[:destination_url] = path(topic.relative_url) if topic.present?
+      if user.present?
+        if user.active?
+          if user.guardian.can_see?(topic)
```

Comment:
Is this because the topic's permission changed after the invite has been sent? Seems strange the we don't give the user  permission to view the topic when the user has been invited from the topic.

Reply:
It is possible to invite a user to a topic they cannot see.

The most obvious case is when you invite a user to a topic that only a group can see, but you do not invite the user to that group too. I am handling this by adding a warning in a different PR.

Then, there are the most complex cases when the topic is moved or permissions change and if the user redeems the invite later then they were welcomed by a 404 page.

Reply:
So where does the invitee get sent to, then? The homepage as though no topic had been specified?